src/services/OlCoordinateService.js

```diff
@@ -212,4 +214,61 @@ export class OlCoordinateService {
 		const coordinates = coordinateLikeAsArray.map((c) => c.slice(0, 2));
 		return singleValue ? coordinates[0] : coordinates;
 	}
+
+	/**
+	 * Calculates the length of an array of coordinates.
+	 *
+	 * Coordinates with a global coordinateRepresentation will be always result in geodetic distances (spherical).
+	 * @param {Array<module:domain/coordinateTypeDef~Coordinate>} coordinates input coordinates
+	 * @param {module:domain/coordinateRepresentation~CoordinateRepresentation}  [coordinateRepresentation=GlobalCoordinateRepresentations.WGS84] the coordinateRepresentation of the coordinates
+	 */
+	getLength(coordinates, coordinateRepresentation = GlobalCoordinateRepresentations.WGS84) {
+		const getGeodesicLength = (coordinates) => {
+			const wgs84 = Geodesic.WGS84;
+			return coordinates.reduce((sum, current, index, coordinates) => {
+				if (index === coordinates.length - 1) {
+					return sum;
+				}
+				const next = coordinates[index + 1];
+				const r = wgs84.Inverse(current[1], current[0], next[1], next[0]);
+
+				return sum + r.s12;
+			}, 0);
+		};
+		const wgs84Coordinates =
+			coordinateRepresentation?.code && coordinateRepresentation?.code !== 4326
```

Comment:
`coordinateRepresentation?.code && coordinateRepresentation?.code !== 4326`

- unneeded check for code 4326 (ol.transform can safely always be called even if the coordinates are already in the target SRID)
- I would suggest throwing an error when `CoordinateRepresentation` contains no `srid` property

Reply:
ok, so....

- coordinateRepresentation is not optional
- and something like `if(!coordinateRepresentation.code) throw new Error('The coordinates must have a reprojectable CoordinateRepresentation (code!== null)')`